module/VuFind/src/VuFind/Http/ServerUrlHelper.php

```diff
@@ -0,0 +1,80 @@
+<?php
+
+/**
+ * Server URL Helper class.
+ *
+ * PHP version 8
+ *
+ * Copyright (C) Villanova University 2025.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, see
+ * <https://www.gnu.org/licenses/>.
+ *
+ * @category VuFind
+ * @package  Http
+ * @author   Demian Katz <demian.katz@villanova.edu>
+ * @author   Maccabee Levine <msl321@lehigh.edu>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org
+ */
+
+namespace VuFind\Http;
+
+use Closure;
+use Laminas\View\Renderer\PhpRenderer;
+
+/**
+ * Server URL Helper class.  Wrapper around Laminas ServerUrlHelper.
+ *
+ * @category VuFind
+ * @package  Http
+ * @author   Demian Katz <demian.katz@villanova.edu>
+ * @author   Maccabee Levine <msl321@lehigh.edu>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org
+ */
+class ServerUrlHelper
+{
+    /**
+     * Server URL helper
+     */
+    protected Closure $serverUrlHelper;
+
+    /**
+     * Constructor.
+     *
+     * @param PhpRenderer $viewRenderer View renderer
+     *
+     * @return void
+     */
+    public function __construct(PhpRenderer $viewRenderer)
+    {
+        $this->serverUrlHelper = Closure::fromCallable($viewRenderer->plugin('serverurl'));
```

Comment:
I would actually move this `Closure::fromCallable` logic into the factory, and make this helper an even thinner wrapper. Then you could use constructor property promotion to make things even shorter and simpler. I don't think there's any reason to pass in a whole PhpRenderer if we're just using it to extract a helper -- that's factory work! :-)